src/main/java/com/maxmind/db/NodeCache.java

```diff
@@ -2,12 +2,36 @@
 
 import java.io.IOException;
 
+/**
+ * NodeCache is an interface for a cache that stores decoded nodes.
```

Comment:
Unfortunately `NodeCache` is a bit of a misnomer (or at least inconsistent with other uses of "node"). Maybe we could change this to something like:

```suggestion
 * NodeCache is an interface for a cache that stores decoded values from the data section of the database.
```